odoo/addons/base/tests/common.py

```diff
@@ -8,6 +8,27 @@
 from odoo import Command
 
 
+class PerformanceCommon(TransactionCase):
+
+    def assertModelCreateMulti(self, model_name, vals_list=None):
+        """Ensures model_name records can be created in batch.
+
+        :param str model_name: name of the model on which batch record creation should be verified
+        :param list vals_list: example list of values, if needed
+        """
+        vals_list = vals_list or [dict(), dict()]
+        if not isinstance(vals_list, list):
+            raise ValueError("Invalid arguments for assertModelCreateMulti")
+        assert len(vals_list) > 1, "Batch creation cannot be tested with only one value."
+        def patched_create(self, values_list):
+            if len(values_list) != len(vals_list):
+                self.fail(f"Model {model_name} doesn't support batch records creation.")
+            # Do not create anything
+            return self
```

Comment:
Never write a test on correct code, because it hides the fact that the code of your test is incorrect: `self.fail()` does not exist, since `self` is not a test case, but a recordset.  You should always check that the test fails as expected first.

BTW I suggest to use a mockup object that returns some predefined value (like some existing record(s)).
The test can simply check how many times it has been called and/or with what parameters.

Reply:
Indeed, my bad for the self.fail :facepalm:.
I'll see if I can add a test verifying the util.
Why can't we just return an empty recordset, we only want to make sure the values are still in batch, there is no need to test anything else or return dummy values :thinking:.